File: Array/49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for(String s: strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedString = new String(charArray);
            map.putIfAbsent(sortedString, new ArrayList<>());
            List<String> group = map.get(sortedString);
            group.add(s);
        }

        return new ArrayList<>(map.values());


    }
}

File: Array/49-group-anagrams/group-anagrams.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = defaultdict(list)

        for s in strs:
            char_list = list(s)
            char_list.sort()
            sorted_s = "".join(char_list)
            groups[sorted_s].append(s)

            # groups["".join(sorted(s))].append(s)
        
        return list(groups.values())

